Makefile.work

```diff
@@ -415,87 +498,75 @@ SONIC_BUILD_INSTRUCTION :=  make \
 
 .PHONY: sonic-slave-build sonic-slave-bash init reset
 
+
+ifeq ($(filter clean,$(MAKECMDGOALS)),)
+COLLECT_BUILD_VERSION = { DBGOPT='$(DBGOPT)' scripts/collect_build_version_files.sh $$?; }
+endif
```

Comment:
@Kalimuthu-Velappan 

`make clean` is broken now because of these changes:
```
bash: -c: line 1: syntax error near unexpected token `;'
bash: -c: line 1: `make -f slave.mk PLATFORM= PLATFORM_ARCH=amd64 
....
MIRROR_URLS= MIRROR_SECURITY_URLS=  Q=@ clean; ; '
make[1]: *** [Makefile.work:530: clean] Error 2
```
If COLLECT_BUILD_VERSION is not set we get double ;; in 
`bash -c "$(FIX_DOCKER_GROUP); $(SONIC_BUILD_INSTRUCTION) $@; $(COLLECT_BUILD_VERSION); $(SLAVE_SHELL)"`

Reply:
Also I'm not fully understand why we need this specifc handling for `clean` command.
I mean why we filter only `clean` here, but not `distclean` e.g.